src/picongpu/include/plugins/adios/ADIOSWriter.hpp

```diff
@@ -586,131 +793,102 @@ class ADIOSWriter : public ILightweightPlugin
-     */
-    static void defineMetaAttributes(ThreadParams *threadParams)
-    {
         AdiosUInt32Type adiosUInt32Type;
         AdiosFloatXType adiosFloatXType;
         AdiosDoubleType adiosDoubleType;
 
-        /* iteration, sim_slides */
-        defineMetaAttr(threadParams, "iteration", adiosUInt32Type.type);
-        defineMetaAttr(threadParams, "sim_slides", adiosUInt32Type.type);
-        threadParams->adiosGroupSize += 2 * sizeof(uint32_t);
-
-        /* normed grid parameters */
-        defineMetaAttr(threadParams, "delta_t", adiosFloatXType.type);
-        defineMetaAttr(threadParams, "cell_width", adiosFloatXType.type);
-        defineMetaAttr(threadParams, "cell_height", adiosFloatXType.type);
-        if (simDim == DIM3)
-            defineMetaAttr(threadParams, "cell_depth", adiosFloatXType.type);
-        threadParams->adiosGroupSize += (1 + simDim) * sizeof(float_X);
-
-        /* base units*/
-        defineMetaAttr(threadParams, "unit_energy", adiosDoubleType.type);
-        defineMetaAttr(threadParams, "unit_length", adiosDoubleType.type);
-        defineMetaAttr(threadParams, "unit_speed", adiosDoubleType.type);
-        defineMetaAttr(threadParams, "unit_time", adiosDoubleType.type);
-        defineMetaAttr(threadParams, "unit_mass", adiosDoubleType.type);
-        defineMetaAttr(threadParams, "unit_charge", adiosDoubleType.type);
-        defineMetaAttr(threadParams, "unit_efield", adiosDoubleType.type);
-        defineMetaAttr(threadParams, "unit_bfield", adiosDoubleType.type);
-        threadParams->adiosGroupSize += 8 * sizeof(double);
-
-        /* physical constants */
-        defineMetaAttr(threadParams, "mue0", adiosFloatXType.type);
-        defineMetaAttr(threadParams, "eps0", adiosFloatXType.type);
-        threadParams->adiosGroupSize += 2 * sizeof(float_X);
-    }
-
-    /**
-     * Write a single scalar meta attribute
-     */
-    static void writeMetaAttr(ThreadParams *threadParams, void *var)
-    {
-        int64_t var_id = threadParams->adiosMetaAttrVarIds.back();
-        threadParams->adiosMetaAttrVarIds.pop_back();
-
-        ADIOS_CMD(adios_write_byid(threadParams->adiosFileHandle, var_id, var));
-    }
-
-    /**
-     * Write meta attributes
-     * Attributes must be written in same order as defined using \see defineMetaAttributes
-     *
-     * @param threadParams parameters
-     */
-    static void writeMetaAttributes(ThreadParams *threadParams)
-    {
         /* write number of slides to timestep in adios file */
         uint32_t slides = MovingWindow::getInstance().getSlideCounter(threadParams->currentStep);
 
-        float_X varFloatX;
-        double varDouble;
+        //float_X varFlX;
+        //double  varDbl;
 
         /* write current iteration */
-        writeMetaAttr(threadParams, &(threadParams->currentStep));
+        log<picLog::INPUT_OUTPUT > ("ADIOS: meta: iteration");
+        //ADIOS_CMD(adios_common_define_attribute_byvalue(threadParams->adiosGroupHandle,
```

Comment:
that's an upcoming feature for the next adios release. I would like to keep the reminder - I'll put a comment above.